### PR TITLE
2d analysis test

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/cheminfo/nmrium/issues/1708

### CODE REVIEW

**Comment by lpatiny** [comment]
@wadjih-bencheikh18  In general when we check for a SVG element, could you add a condition on the length of this element (length greater than XXX). The goal is to check that if suddenly the SVG contains only one segment instead of a huge path we can detect it. Maybe as minimal length the current length of the element * 80%. You may also add this kind of condition in other tests you did.

Otherwise it looks good to me !

File: test-e2e/panels/spectra.test.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ test('Check change spectrum color, Should be white', async ({ page }) => {
   await expect(whiteSpectrumLine).toBeVisible();
 });
 
-test('Should 2d deactivate spectrum', async ({ page }) => {
+test.skip('Should 2d deactivate spectrum', async ({ page }) => {
```
Comment (lpatiny) [inline test-e2e/panels/spectra.test.ts:92]: You forgot to remove the '.skip'. There is also the possibility to focus on a test using '.only'. Could be easier than adding skip everywhere.

Comment (wadjih-bencheikh18) [reply test-e2e/panels/spectra.test.ts:92]: yes sorry, thank youu

File: test-e2e/NmriumPage.ts [hunk 2]
```diff
@@ -76,4 +76,13 @@ export default class NmriumPage {
       this.page.locator('_react=FilterPanel >> text=Phase correction'),
     ).toBeVisible();
   }
+  public async checkSVGLength() {
```
Comment (lpatiny) [inline test-e2e/NmriumPage.ts:79]: The idea here is still to hardcode in the test a minimal value and not base it on bodyLength.

This means that you need to first time to make a console.log to get the length of the SVG and then hardcode this value.

We could consider that the test pass if the value is >0.8 * expected value and <1.2 * expected value.

File: test-e2e/panels/spectra.test.ts [hunk 3]
```diff
@@ -106,3 +106,257 @@ test('Should 2d deactivate spectrum', async ({ page }) => {
...
+    ).toBeVisible();
+    // Check that spectrum color changed
+    await expect(
+      nmrium.page.locator(
+        'data-test-id=spectrum-line >> _react=Line[display.color="#ddb1c9ff"]',
+      ),
+    ).toBeVisible();
+
+    // Check svg
+    await nmrium.checkSVGLength();
```
Comment (lpatiny) [inline test-e2e/panels/spectra.test.ts:159]: Need to add an hardcoded value there

Comment (wadjih-bencheikh18) [reply test-e2e/panels/spectra.test.ts:159]: okey i get it thanks

**Pull request merged**